Update index.md for the JavaScript Language Overview (#28671)
mdn/content: files/en-us/web/javascript/language_overview/index.md

- [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)-based
- [`async`](/en-US/docs/Web/JavaScript/Reference/Statements/async_function)/[`await`](/en-US/docs/Web/JavaScript/Reference/Operators/await), which is a syntactic sugar for Promises

> For example, here's how a file-read operation may look in JavaScript:

Reviewer: ```suggestion
For example, here's what a file-read operation might look like in JavaScript:
```
